azure/src/main/scala/quasar/physical/blobstore/azure/AzureDatasourceModule.scala

```diff
@@ -51,7 +50,7 @@ object AzureDatasourceModule extends LightweightDatasourceModule {
   override def lightweightDatasource[F[_]: ConcurrentEffect: ContextShift: MonadResourceErr: Timer](
       json: Json)(
       implicit ec: ExecutionContext)
-      : F[InitializationError[Json] \/ Disposable[F, DS[F]]] = {
+      : Resource[F, Either[InitializationError[Json], DS[F]]] = {
```

Comment:
Any particular reason of changing to this signature, rather than a more 1-1 translation like `F[Either[InitializationError[Json], Resource[F, DS[F]]]]`?

Reply:
The previous signature was required since `Disposable` does not manage resource allocation, just deallocation, `Resource` handles both. Even if _datasource_ allocation results in an error (maybe due to validation or environmental issues), there still may be dependent resources that require deallocation, so we want to encapsulate the entire process in `Resource`.